TikTokTrainer/Data/DataProcessing.swift

```diff
@@ -69,23 +73,31 @@ struct ScoringFunction {
     }
     
     
-    private func computeAngleDifferences(preRecordedVid: ProcessedVideo, recordedVid: ProcessedVideo) -> [[CGFloat]] {
+    private func computeAngleDifferences(preRecordedVid: ProcessedVideo, recordedVid: ProcessedVideo) -> [[String: CGFloat]] {
         let preRecordedPoses = computeAngles(video: preRecordedVid)
         let recordedPoses = computeAngles(video: recordedVid)
         let minSlices = min(preRecordedVid.data.count, recordedVid.data.count)
         
-        var angleDifferences = [[CGFloat]](
-            repeating: [CGFloat](),
-            count: minSlices
-        )
+        var angleDifferences = [[String: CGFloat]]()
         
         for (row, poseAngles) in preRecordedPoses.enumerated() where row < minSlices {
+            let slicesToCheck = 3
+            // this also works well, need bigger sample size
+            // let slicesToCheck = 10
+            var sliceData = [String: CGFloat]()
             for (_, angle) in poseAngles.enumerated() {
-                // This is where to add shifts or padding to angle differences
-                // i.e. we can ignore 3 degree differences in the angle by subtracting 3 from the abs(...)
-                angleDifferences[row].append(abs(angle.value - recordedPoses[row][angle.key]!))
+                var lowestSliceScore = abs(angle.value - recordedPoses[row][angle.key]!)
+                for possibleSlice in (row < slicesToCheck ? 0 : row - slicesToCheck)...(row + slicesToCheck > minSlices - 1 ? minSlices - 1 : row + slicesToCheck) {
+                    var sliceScore = abs(preRecordedPoses[possibleSlice][angle.key]! - recordedPoses[possibleSlice][angle.key]!)
+                    if sliceScore < lowestSliceScore {
+                        lowestSliceScore = sliceScore
+                    }
+                }
+                sliceData[angle.key] = lowestSliceScore
```

Comment:
This version of the implementation looks good, but this is more of a philosophical question of how we want to do the best slice score: Do we want to compare each individual angle to the same angle before and after the current slice, so that in one error computation we have angles from different slices potentially? Or do we want to keep all of the angles from a single slice, which entails just comparing the total angle error from 3 slices before and after the current slice and picking the lowest one of that? This isn't meant to prevent the PR, so you can close it after it's read and we can talk about it later.